### ISSUE TITLE
Nav: Add NavFollow component

### ISSUE THREAD

**Issue by reeseschultz** [reeseschultz/ReeseUnityDemos#53]
**Is the improvement related to a problem? Please describe.**

Yes, there's no way to make one agent follow another easily.

**Describe the solution you'd prefer:**

Adding a `NavFollow` component with a `Target` property of type `Entity` that allows one agent to follow another.

**Describe alternatives or workarounds you've considered:**

There's no reasonable workaround to accomplish this.

**Additional context:**

Following should be much easier to implement in the `feat/demo-overhaul` branch, meaning it is dependent on #48, which is a huge, over-scoped issue.

**Comment by reeseschultz** [comment]
I made #48 specifically about merging the Stranded demo, making it much closer to be ready to merge into `master`. I also made sure the feature's branch, `feat/demo-overhaul`, is up-to-date with `master` today.

That expedites the addition of a `NavFollow` component as much as possible @P-Stringer. I'm trying to get to it. There's still more to do for #48 before I can merge it into `master`, though.

**Comment by reeseschultz** [comment]
https://user-images.githubusercontent.com/43183393/106209205-9d78ef00-6192-11eb-8a58-a9f736e30d00.mp4

@P-Stringer I added `NavFollow` and `NavFollowAuthoring`. It's all in `master` (`nav 0.19.0`). Any `NavAgent` with a `NavFollow` component will be processed by the built-in `NavFollowSystem`. `NavAgentHybrid` GameObjects should have support too. It's all documented in the [nav README](https://github.com/reeseschultz/ReeseUnityDemos/tree/nav). You can see it in action in the `Stranded` demo as well, where someone wearing a banana costume follows the player just because.

You'll want to take a look not only at the `Target` property, but also `MaxDistance` and `MinDistance`.

Hope this helps.

**Comment by P-Stringer** [comment]
Fanatastic! Looks really good @reeseschultz :) Was meant to leave a comment about adding distance but never got round to it - thanks a lot, I'll give it a go.
